src/lines_lf.rs

```diff
@@ -47,51 +47,62 @@ pub fn to_byte_idx(text: &str, line_idx: usize) -> usize {
 }
 
 //-------------------------------------------------------------
+const LF: u8 = b'\n';
 
 #[inline(always)]
 fn to_byte_idx_impl<T: ByteChunk>(text: &[u8], line_idx: usize) -> usize {
+    let mut byte_count = 0;
+    let mut lf_count = 0;
+
+    if text.len() <= T::SIZE {
+        // Bypass the more complex routine for short strings, where the
+        // complexity hurts performance.
```

Comment:
Does this have a significant performance impact?  If it's only a minor perf improvement, I'd rather leave it out to keep the code simpler.